hayeon/programmers-2/이진-변환-반복하기.js

```diff
@@ -0,0 +1,43 @@
+function solution(s) {
+  let [translationCount, ZeroCount] = [0, 0];
+
+  while (s !== "1") {
+    const originLength = s.length;
+
+    s = s.replace(/0/g, "");
+    ZeroCount += originLength - s.length;
+
+    s = s.length.toString(2);
+    translationCount++;
+  }
+  return [translationCount, ZeroCount];
+}
```

Comment:
```suggestion
function solution(s) {
  let [translationCount, zeroCount] = [0, 0];

  while (s !== "1") {
    const originLength = s.length;

    s = s.replace(/0/g, "");
    zeroCount += originLength - s.length;

    s = s.length.toString(2);
    translationCount++;
  }
  return [translationCount, zeroCount];
}
```
- 풀이 잘봤씁니다 ~~!
- 오노우 ZeroCount 파스켈 케이스로 네이밍이 작성되어있어요! zeroCount 로 변경해주세요~
- 나머지는 훌륭하십니다 고생하셨어요!
- 아래와 같이 replace 2번째 인자로 콜백 함수를 넣어서도 해결하실 수 있습니다.
```js
function solution(s) {
  const result = [0, 0];

  while (s.length > 1) {
    const convertedStr = s.replace(/0/g, () => {
      result[1]++; // 제거된 0의 개수 카운팅
      return "";
    });

    s = convertedStr.length.toString(2);
    result[0]++; // 변환 횟수 카운팅
  }

  return result;
}
```

Reply:
- 앗.. 수업하면서 자꾸 앞에 대문자로 써서 무의식중에 그렇게 썼나봐요 죄송합니다🥺
- 콜백으로도 가능하군여!!!!!!! 참고해보겠습니다ㅎㅎ 감사합니다!!🙇‍♀️